src/Microsoft.Extensions.DiagnosticAdapter/Internal/ProxyTypeEmitter.cs

```diff
@@ -101,8 +101,11 @@ private static bool VerifyProxySupport(ProxyBuilderContext context, Tuple<Type,
             }
 
             ProxyTypeCacheResult cacheResult;
+            var verificationResult = new VerificationResult();
             if (context.Cache.TryGetValue(key, out cacheResult))
             {
+                context.Visited.Add(key, verificationResult);
```

Comment:
I think we also need to copy the related properties from `catchResult` -> `verificationResult`